File: Cinema_groupe4_back/cinema_backend/src/main/java/com/intiformation/modele/Cinema.java
package com.intiformation.modele;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Cinema {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long idCinema;
	private String nom;
	private long telephone;
	private String mail;
	private String rue;
	private String ville;
	private int codePostal;

	public Cinema() {
	}

	public Cinema(String nom, long telephone, String mail, String rue, String ville, int codePostal) {
		this.nom = nom;
		this.telephone = telephone;
		this.mail = mail;
		this.rue = rue;
		this.ville = ville;
		this.codePostal = codePostal;
	}

	public Cinema(long idCinema, String nom, long telephone, String mail, String rue, String ville, int codePostal) {
		this.idCinema = idCinema;
		this.nom = nom;
		this.telephone = telephone;
		this.mail = mail;
		this.rue = rue;
		this.ville = ville;
		this.codePostal = codePostal;
	}

	public long getIdCinema() {
		return idCinema;
	}

	public void setIdCinema(long idCinema) {
		this.idCinema = idCinema;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public long getTelephone() {
		return telephone;
	}

	public void setTelephone(long telephone) {
		this.telephone = telephone;
	}

	public String getMail() {
		return mail;
	}

	public String getRue() {
		return rue;
	}

	public String getVille() {
		return ville;
	}

	public int getCodePostal() {
		return codePostal;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}

	public void setRue(String rue) {
		this.rue = rue;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	public void setCodePostal(int codePostal) {
		this.codePostal = codePostal;
	}

}

File: Cinema_groupe4_front/cinema_front/src/app/model/cinema.ts
export class Cinema {

    private _idCinema : number;
    private _nom : string;
    private _telephone : number;
    private _mail : string;
    private _rue : string;
    private _ville : string;
    private _codePostal : number;

    get idCinema() : number {
        return this._idCinema;
    }

    set idCinema(value : number) {
        this._idCinema = value;
    }

    get nom() : string {
        return this._nom;
    }

    set nom(value : string) {
        this._nom = value;
    }

    get telephone() : number {
        return this._telephone;
    }

    set telephone(value : number) {
        this._telephone = value;
    }

    get mail() : string {
        return this._mail;
    }

    set mail(value : string) {
        this._mail = value;
    }

    get rue() : string {
        return this._rue;
    }

    set rue(value : string) {
        this._rue = value;
    }

    get ville() : string {
        return this._ville;
    }

    set ville(value : string) {
        this._ville = value;
    }

    get codePostal() : number {
        return this._codePostal;
    }

    set codePostal(value : number) {
        this._codePostal = value;
    }

    public constructor (idCinema : number, nom : string,telephone : number,mail : string, rue : string, ville : string, codePostal : number){
        this._idCinema = idCinema;
        this._nom= nom;
        this._telephone = telephone;
        this._mail = mail;
        this._rue = rue;
        this._ville = ville;
        this._codePostal = codePostal;
    }
}